tool/tsh/kube.go

```diff
@@ -912,6 +911,11 @@ func buildKubeConfigUpdate(cf *CLIConf, kubeStatus *kubernetesStatus) (*kubeconf
 	}
 
 	if len(kubeStatus.kubeClusters) == 0 {
+		if cf.IdentityFileOut == "" {
```

Comment:
Why is this only when identity file path is empty?

Reply:
This is a case when  `v.Exec == nil ` when  `buildKubeConfigUpdate` 
 was called by tsh flow the current flow creates the `KUBECONF` file with default creds without dependency to tsh binary.  I was avoiding changing this behaviour. 
 
 https://github.com/gravitational/teleport/blob/59a72c87e8aa85ed9861146f4d608ef7933158ce/lib/kube/kubeconfig/kubeconfig.go#L152

Reply:
Got it, in that case could you update the comment below to explain that (so explain "why" rather than "what")? Otherwise it is not clear without looking at kubeconfig.Update implementation.